pkg/ruler/rule_concurrency.go

```diff
@@ -135,8 +139,8 @@ type TenantConcurrencyController struct {
 // Done releases a slot from the concurrency controller.
 func (c *TenantConcurrencyController) Done(_ context.Context) {
 	c.globalConcurrency.Release(1)
-	c.metrics.CurrentConcurrency.Dec()
-	c.metrics.ReleaseTotal.Inc()
+	c.metrics.SlotsInUser.WithLabelValues(c.tenantID).Dec()
```

Comment:
[nit] We may store the per-tenant metric directly in the `TenantConcurrencyController` struct, so that we don't have to call `WithLabelValues()` each time. There's a small perf overhead calling `WithLabelValues()` which we could simply avoid.

Reply:
Done in [1e6a32a](https://github.com/grafana/mimir/pull/8884/commits/1e6a32ad76d9a3776402d6bd744c96dd736d643b)